packages/server/src/middleware/cleanup.ts

```diff
@@ -0,0 +1,16 @@
+import { Ctx } from "@budibase/types"
+import { context } from "@budibase/backend-core"
+
+export default async (ctx: Ctx, next: any) => {
+  const resp = next()
+
+  const current = context.getCurrentContext()
+  if (current?.cleanup) {
+    for (let fn of current.cleanup || []) {
+      await fn()
```

Comment:
should probably wrap this in a `try catch` - just incase any error is thrown during the cleanup, we probably want to finish cleaning up other stuff, and alert in someway that a cleanup failed.